src/main/java/com/khutircraftubackend/auth/request/RegisterRequest.java

```diff
@@ -14,6 +15,8 @@ public record RegisterRequest(
         String password,
 
         @NotNull(message = "Роль не може бути порожньою")
-        Role role
+        Role role,
+
+        SellerDTO details
```

Comment:
it's ok now, 

but might be a problem in case if business decides to introduce a new role that allows to get registered with user details, example: DeliveryPartner 

is it possible to make details more generic, or have it as a Map that could be programmatically mapped to some DTO ?

Reply:
you can add fields to the details, and then take those that are needed for each role. At the same time, do not put not null in the database